src/main/java/de/vogel612/testclient_javabot/client/ClientGui.java

```diff
@@ -1,156 +1,92 @@
-        this.add(new JLabel("Testing site for JavaBot aka. Junior"), c);
-    }
-
-    private void setupMessageTable() {
-        messageTable.setAutoResizeMode(JTable.AUTO_RESIZE_LAST_COLUMN);
-        messageTable.setAutoCreateColumnsFromModel(true);
-        messageTable.setFillsViewportHeight(true);
-        messageTable.setDefaultRenderer(ChatMessage.class, new ChatCellRenderer());
-        messageTable.setVisible(true);
-    }
-
-    private void bindListeners() {
-        userInput.addKeyListener(new InputListener());
-        submit.addActionListener(action -> {
-            chatClient.newUserMessage(userInput.getText());
-            userInput.setText("");
-            userInput.requestFocus();
-        });
-    }
-
-    @Override
-    public void start() {
-        this.setVisible(true);
-    }
-
-    @Override
-    public boolean enqueueMessage(final ChatMessage chatMessage) {
-        tableModel.addNewMessage(chatMessage);
-        return true;
-    }
-
-    private final class InputListener implements KeyListener {
-
-        @Override
-        public void keyPressed(KeyEvent ev) {
-            if (ev.getKeyCode() == KeyEvent.VK_ENTER) {
-                if (ev.isShiftDown()) {
-                    // make a newline for it
-                    userInput.append("\r\n");
-                    return;
+import de.vogel612.testclient_javabot.client.controller.ChatRenderController;
+
+public class ClientGui implements ChatWorker {
+	
+	private static final Logger LOGGER = Logger.getLogger(ClientGui.class.getName());
+	
+	private ChatRenderController controller;
+	private JFrame frame;
+	final CountDownLatch latch = new CountDownLatch(1);
+
+	public void initFX(JFXPanel fxPanel) throws Exception {
+		FXMLLoader loader = new FXMLLoader(getClass().getResource(
+	               "/de/vogel612/testclient_javabot/client/fxml/ChatRender.fxml"));
+		BorderPane borderPane = (BorderPane)loader.load();
+		controller = loader.getController();
+		Scene scene = new Scene(borderPane);
+		scene.getStylesheets().add(getClass().getResource("/de/vogel612/testclient_javabot/client/style/style.css").toExternalForm());
+		// Remove this to disable the dark theme
+		scene.getStylesheets().add(getClass().getResource("/de/vogel612/testclient_javabot/client/style/darkTheme.css").toExternalForm());
+		fxPanel.setScene(scene);
+	}
+	
+	public ClientGui() {
+		try{
+			SwingUtilities.invokeLater(new Runnable() {
+	            @Override
+	            public void run() {
+	            	frame = new JFrame("TestClient for Junior");
+	                final JFXPanel fxPanel = new JFXPanel();
+	                frame.add(fxPanel);
+	                frame.setSize(550, 421);
+	                frame.setVisible(true);
+	                frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+	
+	                Platform.runLater(new Runnable() {
+	                    @Override
+	                    public void run() {
+	                        try {
+								initFX(fxPanel);
```

Comment:
Is this indentation wanted?
Also.. why not use java 8 features and drop the inner class in favor of a lambda?